source/cl/test/UnitCL/kernels/regression.109_libm_native_double_input.cl

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) Codeplay Software Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License") with LLVM
+// Exceptions; you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/LICENSE.txt
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// CLC OPTIONS: -cl-fast-relaxed-math
+// SPIRV OPTIONS: -cl-fast-relaxed-math
+
+// The purpose of this test is to exercise builtin native functions where we use
+// half as input. We use the same logic as other testing of builtin functions
+// where we give opportunity to be replaced by LLVM intrinsics, which may in
+// turn be replaced by calls to libm.  That can either only happen when libm is
+// accessible (e.g. by the loader linking in libm functions).
+//
+// This test specifically requires fast-math because that gives the compiler
+// more lee-way to do the replacement.
+
+// To avoid needing to account for ULP errors in the various functions being
+// tested, just compare the result against an impossible value (for the given
+// inputs, which the compiler does not know).  For the trig functions a smart
+// compiler could maybe figure out that the comparison is impossible, but at
+// the time of writing ComputeAorta is not that smart.
+
+// REQUIRES: double
+
+#define BIG_VALF 5.0f
```

Comment:
Should the literal constants in this test not have suffix `h` because it says REQUIRES double?

Reply:
That said, there are some unnecessary implicit conversions elsewhere (e.g. `native_exp10(in[index]) >= BIG_VALF * 3.0` and the `... ? 2.0f : 1.0f` in this file), and I am not sure I understand the reasoning for returning a `double` that can only be `1.0` or `2.0`.

Reply:
yes, it's a bit weird, I copied from another regression test but it seems unnecessary. I've changed to 1/2 as uint.

Reply:
Created the suggested edits for the remaining implicit conversions.

Reply:
thanks